lib/ansible/module_utils/tacp.py

```diff
@@ -246,12 +248,20 @@ def wait_for_action_to_complete(action_uuid, api_client):
         api_instance = tacp.ActionsApi(api_client)
         action_incomplete = True
 
-        while action_incomplete:
+        timeout = 30
+        time_spent = 0
+
+        while action_incomplete and time_spent < timeout:
             api_response = api_instance.get_action_using_get(action_uuid)
             if api_response.status == 'Completed':
                 action_incomplete = False
+                return True, None
             else:
                 sleep(1)
+                time_spent += 1
+
+        return False, "Action timed out."
```

Comment:
Instead of returning this tuple, I would recommend creating and raising a custom exception. This exception can be caught and handled by the caller.